ui/lib/core/addon/components/form-field.hbs

```diff
@@ -5,409 +5,445 @@
 
 {{! template-lint-configure simple-unless "warn" }}
 <div class="field" data-test-field={{or @attr.name true}}>
-  {{#unless this.hideLabel}}
-    <FormFieldLabel
-      for={{@attr.name}}
-      @label={{this.labelString}}
-      @helpText={{(if this.showHelpText @attr.options.helpText)}}
-      @subText={{@attr.options.subText}}
-      @docLink={{@attr.options.docLink}}
-    />
-  {{/unless}}
-  {{#if @attr.options.possibleValues}}
-    {{#if (eq @attr.options.editType "radio")}}
-      <div class="control {{unless this.hasRadioSubText 'columns'}}" data-test-radio-input>
-        {{#each (path-or-array @attr.options.possibleValues @model) as |val|}}
-          <div
-            class="is-flex-center
-              {{if this.hasRadioSubText 'has-top-padding-xs has-bottom-padding-s' 'column is-narrow has-right-margin-s'}}"
-            data-test-input={{@attr.name}}
-          >
-            <RadioButton
-              class="radio"
-              name={{@attr.name}}
-              id={{or val.value val}}
-              value={{or val.value val}}
-              @value={{or val.value val}}
-              @onChange={{this.setAndBroadcast}}
-              @groupValue={{get @model this.valuePath}}
-              @disabled={{and @attr.options.editDisabled (not @model.isNew)}}
-              data-test-radio={{or val.value val}}
-            />
-            <div class="has-left-margin-xs">
-              <label
-                for="{{or val.value val}}"
-                class="has-left-margin-xs is-size-7"
-                data-test-radio-label={{or val.label val.value val}}
-              >
-                {{or val.label val.value val}}
-                {{#if val.helpText}}
-                  <Hds::TooltipButton @text={{val.helpText}} aria-label="More information">
-                    <Hds::Icon @name="info" @isInline={{true}} />
-                  </Hds::TooltipButton>
-                {{/if}}
-                {{#if this.hasRadioSubText}}
-                  <p class="has-left-margin-xs has-text-grey is-size-8" data-test-radio-subText={{val.subText}}>
-                    {{val.subText}}
-                  </p>
+  {{#if this.isHdsFormField}}
+    {{#if (or (eq @attr.type "number") (eq @attr.type "string"))}}
+      {{#if (eq @attr.options.editType "password")}}
+        <Hds::Form::TextInput::Field
+          @type="password"
+          {{! TODO discuss with Vault team what they do prefer to do here }}
+          @hasVisibilityToggle={{false}}
+          @value={{get @model this.valuePath}}
+          name={{@attr.name}}
+          @isInvalid={{this.validationError}}
+          {{! Prevents browsers from auto-filling }}
+          autocomplete="new-password"
+          spellcheck="false"
+          data-test-input={{@attr.name}}
+          as |F|
+        >
+          {{#unless this.hideLabel}}
```

Comment:
You may already be considering this, but we should avoid carrying `this.hideLabel` logic over to HDS components. Hiding the label was a workaround to support inputs where the label renders beside the input (like a single checkbox) instead of above it. However, each HDS form component handles its own label so we don't need to manage this at the top level anymore. 

Using the checkbox as an example, we can replace the boolean checkbox simply with `<Hds::Form::Checkbox::Field>` instead of hiding/showing the label.

Reply:
Good to know! 👍